python/pyiceberg/catalog/glue.py

```diff
@@ -0,0 +1,453 @@
+        new_input = _construct_table_input(to_table_name, table_input[PROP_GLUE_TABLE_PARAMETERS][METADATA_LOCATION], {})
+
+        try:
+            self.glue.create_table(DatabaseName=to_database_name, TableInput=new_input)
+        except self.glue.exceptions.AlreadyExistsException as e:
+            raise TableAlreadyExistsError(f"Table {to_database_name}.{to_table_name} already exists") from e
+        except self.glue.exceptions.EntityNotFoundException as e:
+            raise NoSuchNamespaceError(f"Database {to_database_name} not found") from e
+
+        try:
+            self.drop_table(from_identifier)
+        except Exception as e:
+            self.drop_table(to_identifier)
+            raise ValueError(
+                f"Fail to drop old table {from_database_name}.{from_table_name}, roll back to use the old one"
+            ) from e
+        return self.load_table(to_identifier)
+
+    def create_namespace(self, namespace: Union[str, Identifier], properties: Properties = EMPTY_DICT) -> None:
+        """Create a namespace in the catalog.
+
+        Args:
+            namespace: Namespace identifier
+            properties: A string dictionary of properties for the given namespace
+
+        Raises:
+            ValueError: If the identifier is invalid
+            AlreadyExistsError: If a namespace with the given name already exists
+        """
+        database_name = self.identifier_to_tuple(namespace)[0]
+        try:
+            self.glue.create_database(DatabaseInput=_construct_database_input(database_name, properties))
+        except self.glue.exceptions.AlreadyExistsException as e:
+            raise NamespaceAlreadyExistsError(f"Database {database_name} already exists") from e
+
+    def drop_namespace(self, namespace: Union[str, Identifier]) -> None:
+        """Drop a namespace.
+
+        Args:
+            namespace: Namespace identifier
+
+        Raises:
+            NoSuchNamespaceError: If a namespace with the given name does not exist, or the identifier is invalid
+            NamespaceNotEmptyError: If the namespace is not empty
+        """
+        database_name = self.identifier_to_tuple(namespace)[0]
+        try:
+            table_list = self.list_tables(namespace=database_name)
+        except NoSuchNamespaceError as e:
+            raise NoSuchNamespaceError(f"Database does not exists: {database_name}") from e
+
+        if len(table_list) > 0:
+            raise NamespaceNotEmptyError(f"Database {database_name} is not empty")
+
+        self.glue.delete_database(Name=database_name)
+
+    def list_tables(self, namespace: Union[str, Identifier]) -> List[Identifier]:
+        database_name = self.identifier_to_tuple(namespace)[0]
+        table_list = []
+        try:
+            table_list_response = self.glue.get_tables(DatabaseName=database_name)
+            next_token = table_list_response.get(PROP_GLUE_NEXT_TOKEN)
+            table_list += table_list_response[PROP_GLUE_TABLELIST]
```

Comment:
Thank you for your suggestion. I think it is a good idea to change it to `.get(PROP_GLUE_TABLELIST, [])`. At the same time, I also do some tests to verify the behavior of the API calls:

According to my test:https://gist.github.com/JonasJ-ap/d75735124951fa11f2bb440cc0b5a9ed, the responses of `get_tables` and `get_databases` will contain `TableList`/`DatabaseList` field even when there is no table/database in the database/aws account. I think there will be no key error in case of empty table list even in the current implementation. 

Moreover, according to the Java's awssdk, the `tableList()` and `databaseList()` will never be `null`. Hence, I think for boto3's corresponding method, `tableList` and `databaseList` field should always exist similar to java impl. The only difference between java's sdk and boto3 is that java's sdk provides an additional method `hasDatabaseList()` to check the internal behavior of the service. 

Please correct me if I misunderstand something. If I am wrong, some additional field checks will be needed in some other parts of this implementation.

Thank you in advance for your help.

Relevant docs:
https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/glue/model/GetDatabasesResponse.html
https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/glue.html#Glue.Client.get_databases